poetry/semver/version_utils.py

```diff
@@ -0,0 +1,48 @@
+"""
```

Comment:
this looks like duplicated code from `semver/constraint_utils.py`.

Reply:
The difference is that this uses `Version.parse` whereas the constraint code uses `parse_single_constraint` and, AFAIK, the constraints are not versions (although versions can be constraints, the distinction stands).